modules/sdk-coin-polygon/src/polygon.ts

```diff
@@ -1,167 +1,20 @@
 /**
  * @prettier
  */
-import { bip32 } from '@bitgo/utxo-lib';
 import request from 'superagent';
-import { ExplainTransactionOptions } from '@bitgo/abstract-eth';
-import {
-  Eth,
-  Recipient,
-  GetSendMethodArgsOptions,
-  SendMethodArgs,
-  optionalDeps,
-  BuildTransactionParams,
-  SignFinalOptions,
-  SignTransactionOptions,
-  SignedTransaction,
-  RecoverOptions,
-  RecoveryInfo,
-  OfflineVaultTxInfo,
-} from '@bitgo/sdk-coin-eth';
-import {
-  BaseCoin,
-  BitGoBase,
-  common,
-  TransactionExplanation,
-  FullySignedTransaction,
-  getIsUnsignedSweep,
-  Util,
-  MPCAlgorithm,
-} from '@bitgo/sdk-core';
+import { AbstractEthLikeNewCoins } from '@bitgo/abstract-eth';
+import { BaseCoin, BitGoBase, common, MPCAlgorithm } from '@bitgo/sdk-core';
 import { BaseCoin as StaticsBaseCoin, coins } from '@bitgo/statics';
-import BigNumber from 'bignumber.js';
-import { KeyPair, TransactionBuilder } from './lib';
-import _ from 'lodash';
-import type * as EthTxLib from '@ethereumjs/tx';
-
-export class Polygon extends Eth {
-  protected readonly _staticsCoin: Readonly<StaticsBaseCoin>;
-  protected readonly sendMethodName: 'sendMultiSig' | 'sendMultiSigToken';
+import { TransactionBuilder } from './lib';
 
+export class Polygon extends AbstractEthLikeNewCoins {
   protected constructor(bitgo: BitGoBase, staticsCoin?: Readonly<StaticsBaseCoin>) {
     super(bitgo, staticsCoin);
-
-    if (!staticsCoin) {
-      throw new Error('missing required constructor parameter staticsCoin');
-    }
-
-    this._staticsCoin = staticsCoin;
   }
 
   static createInstance(bitgo: BitGoBase, staticsCoin?: Readonly<StaticsBaseCoin>): BaseCoin {
     return new Polygon(bitgo, staticsCoin);
   }
-  static getCustomChainName(chainId?: number): string {
-    if (chainId === 80001) {
-      return 'PolygonMumbai';
-    }
-    return 'PolygonMainnet';
-  }
-
-  static buildTransaction(params: BuildTransactionParams): EthTxLib.FeeMarketEIP1559Transaction | EthTxLib.Transaction {
-    // if eip1559 params are specified, default to london hardfork, otherwise,
-    // default to tangerine whistle to avoid replay protection issues
-    const defaultHardfork = !!params.eip1559 ? 'london' : optionalDeps.EthCommon.Hardfork.TangerineWhistle;
-    let customChainCommon;
-    // if replay protection options are set, override the default common setting
-    if (params.replayProtectionOptions) {
```

Comment:
BREAKING CHANGE: replayProtectionOptions is not optional in buildTransaction method for EthLike coins and needs to be passed to derive the Eth common object from the chainId